geode-core/src/main/java/org/apache/geode/management/internal/configuration/functions/DownloadJarFunction.java

```diff
@@ -61,6 +61,13 @@ public void execute(FunctionContext<Object[]> context) {
         RemoteInputStreamServer istream = null;
         try {
           File jarFile = sharedConfig.getPathToJarOnThisLocator(group, jarName).toFile();
+          // if file is modified within the last second, wait for a second before downloading it
+          // It looks like if the file was modified within the same second, the file downloaded
+          // will be the old content. On mac/linux, the lastModified time of the file is already
+          // reported to the precision of seconds.
+          if (jarFile.lastModified() > System.currentTimeMillis() - 1000) {
```

Comment:
This should be replaced with an await so we don't get stuck here forever.

Reply:
I tried that but await isn't available in core.